### ISSUE TITLE
POST/PUT not working with _id of String type

### ISSUE THREAD

**Issue by abhishekvaid** [florianholzapfel/express-restify-mongoose#368]
I'm trying to run canonical example given for this repo with a simple change that _id is of `type: String`. Here's the code I'm trying to run. 

```
const express = require('express');
const bodyParser = require('body-parser');
const methodOverride = require('method-override');
const mongoose = require('mongoose');
const restify = require('express-restify-mongoose');
const app = express();
const router = express.Router();

app.use(bodyParser.urlencoded({ extended: true }));
app.use(bodyParser.json());
app.use(methodOverride());

mongoose.Promise = global.Promise;
mongoose.connect('mongodb://devuser:devpwd@localhost:27017/content_portal_db?authSource=admin', {useMongoClient: true});

restify.serve(router, mongoose.model('Customer', new mongoose.Schema({
    _id: {type: String},
    name: { type: String, required: true },
    comment: { type: String }
})));

app.use(router);

app.listen(8000, () => {
    console.log('Express server listening on port 8000')
});
```

- The server starts okay.
- GET route seems to be working fine: `GET localhost:8000/api/v1/Customer` is working fine. 
- I'm unable to make a PUT/POST request. I don't know what's missing, but trying various options gives me various errors. I'm also unable to debug or display all the routes registered by this module. 

- Here's what's happening:

POST Example with _id in body: [ DOESN'T WORKS ]

<img width="539" alt="screen shot 2017-11-27 at 4 42 11 pm" src="https://user-images.githubusercontent.com/2530932/33264045-f9d46f96-d391-11e7-825a-89218e69faf2.png">

POST Example with _id in url: [ DOESN'T WORKS ]

<img width="448" alt="screen shot 2017-11-27 at 4 41 44 pm" src="https://user-images.githubusercontent.com/2530932/33264055-095fd90a-d392-11e7-8732-8595bb58c714.png">

GET with _id in URL: [WORKS]

<img width="420" alt="screen shot 2017-11-27 at 4 43 47 pm" src="https://user-images.githubusercontent.com/2530932/33264089-3082c8e4-d392-11e7-96a8-f7a6a07fb511.png">

**Comment by Zertz** [comment]
What version are you using?

**Comment by abhishekvaid** [comment]
4.13.5

**Comment by manishrw** [comment]
I am facing the same issue. Any plan on fixes?

**Comment by Zertz** [comment]
I am fairly sure this is related to #326 which is fixed in #358 

That release is extremely close to being ready, it's just a matter of updating docs. I will try to take some time within the next few days to wrap it up.
